crates/emath/src/range.rs

```diff
@@ -49,12 +49,48 @@ impl Rangef {
         self.max - self.min
     }
 
+    /// The length of the range, i.e. `max - min`.
+    #[inline]
+    pub fn center(self) -> f32 {
+        self.min + (self.span() / 2.0)
+    }
+
+    /// Returns the similarity between `self` and `other`.
+    ///
+    ///  * negative if `a` is left of `b`
+    /// * positive if `a` is right of `b`
+    /// * zero if the ranges overlap significantly
+    #[inline]
+    pub fn similarity(self, other: Rangef) -> f32 {
+        if self.contains_range(other) || other.contains_range(self) {
+            0.0
+        } else {
+            -(other.center() - self.center())
+        }
+    }
+
     #[inline]
     #[must_use]
     pub fn contains(self, x: f32) -> bool {
         self.min <= x && x <= self.max
     }
 
+    /// Returns if `self` contains `other` for at least 50%.
+    #[inline]
+    pub fn contains_range(&self, other: Rangef) -> bool {
```

Comment:
This is a very surprising definition of `contains_range`. Either change it to `self.contains(other.min) && self.contains(other.max)`, or move it to where it is being used.

Reply:
Was trying to get the overlap of more than 50 percent to work. So that self looks if it overlaps at least 50% with other. Changing it to self.contains(other.min) && self.contains(other.max) would restrict it to a full overlap where other is fully within self?

Reply:
the 50% overlap may be a good idea for what you are trying to do, but it is not a good idea for a `pub` function called `contains_range`. Call it `has_significant_overlap` or something and move it to where it is used (`crates/egui/src/memory.rs`)

Reply:
the 50% overlap may be a good idea for what you are trying to do, but it is not a good idea for a `pub` function called `contains_range`. Call it `has_significant_overlap` or something and move it to where it is used (`crates/egui/src/memory.rs`)